charts/terra-argocd-app/templates/app.yaml

```diff
@@ -1,39 +1,45 @@
-{{- with .Values -}}
+{{- if .Values.releaseType | eq "app" }}
+# Generates an ArgoCD application for a Terra application release
+{{- $app := required "releaseName is required" .Values.releaseName }}
+{{- $env := required "targetName is required" .Values.targetName }}
 apiVersion: argoproj.io/v1alpha1
 kind: Application
 metadata:
-  name: {{ required "A valid app name is required" .app }}-{{ required "A valid environment is required" .environment }}
+  name: "{{ required "A valid releaseName is required" .Values.releaseName }}-{{ required "A valid targetName is required" .Values.targetName }}"
   labels:
-    app: {{ quote .app }} # These labels are used as selectors in the UI and for argocd CLI calls
-    env: {{ quote .environment }}
-    cluster: {{ quote .clusterName }}
+    app: {{ $app | quote }} # These labels are used as selectors in the UI and for argocd CLI calls
+    release: {{ $app | quote }}
+    env: {{ $env | quote }}
+    cluster: {{ required "A valid clusterName is required" .Values.clusterName | quote }}
     type: app
-    has-legacy-configs: {{ quote .legacyConfigsEnabled }}
-    jenkins-sync-enabled: {{ quote .jenkinsSyncEnabled }}
-  {{- if .purgeDeployedResourcesOnDelete }}
+    has-legacy-configs: {{ .Values.legacyConfigsEnabled | quote }}
+    jenkins-sync-enabled: {{ .Values.jenkinsSyncEnabled | quote }}
+    {{- if .Values.jenkinsSyncGroup }}
+    jenkins-sync-group: {{ .Values.jenkinsSyncGroup | quote }}
+    {{- end }}
+  {{- if .Values.purgeDeployedResourcesOnDelete }}
   finalizers:
   - resources-finalizer.argocd.argoproj.io
   {{- end }}
 spec:
-  project: {{ include "terra-argocd.projectName" . }}
+  project: {{ required "A valid projectName is required" .Values.projectName | quote }}
   destination:
-    namespace: {{ include "terra-argocd.destinationNamespace" . }}
-    server: {{ quote .clusterAddress }}
+    namespace: {{ required "A valid namespace is required" .Values.namespace | quote }}
```

Comment:
Maybe `Release.Namespace` would be useful here?

Reply:
The namespace that is used here (`terra-dev`, etc) is actually different from Release.Namespace (which is `ap-argocd`)

Reply:
Thanks for explaining this!